packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/duck.test.js

```diff
@@ -129,6 +129,102 @@ describe('TraceTimelineViewer/duck', () => {
     });
   });
 
+  describe('expands and collapses all spans', () => {
+    // 0
+    // - 1
+    // --- 2
+    // - 3
+    // --- 4
+    const spans = [
+      { spanID: 0, depth: 0, hasChildren: true },
+      { spanID: 1, depth: 1, hasChildren: true },
+      { spanID: 2, depth: 2, hasChildren: false },
+      { spanID: 3, depth: 1, hasChildren: true },
+      { spanID: 4, depth: 2, hasChildren: false },
+    ];
+
+    const oneSpanCollapsed = new Set([1]);
+    const allSpansCollapsed = new Set([0, 1, 3]);
+    const oneLevelCollapsed = new Set([1, 3]);
+
+    const tests = [
+      {
+        msg: 'expand all',
+        action: expandAll,
+        initial: allSpansCollapsed,
+        resultant: new Set(),
+      },
+      {
+        msg: 'collapse all, no-op',
+        action: collapseAll,
+        initial: allSpansCollapsed,
+        resultant: allSpansCollapsed,
+      },
+      {
+        msg: 'expand one',
+        action: expandOne,
+        initial: allSpansCollapsed,
+        resultant: oneLevelCollapsed,
+      },
+      {
+        msg: 'expand one, one collapsed',
+        action: expandOne,
+        initial: oneSpanCollapsed,
+        resultant: new Set(),
+      },
+      {
+        msg: 'collapse one, no-op',
+        action: collapseOne,
+        initial: allSpansCollapsed,
+        resultant: allSpansCollapsed,
+      },
```

Comment:
Can you add a test case to cover the collapse one scenario going from `[3]` to `[1, 3]`?